src/Database/LSMTree/Internal/Snapshot.hs

```diff
@@ -126,49 +134,68 @@ openLevels ::
   -> HasFS m h
   -> HasBlockIO m h
   -> TableConfig
+  -> UniqCounter m
   -> SessionRoot
+  -> ResolveSerialisedValue
   -> SnapLevels
   -> m (Levels m h)
-openLevels reg hfs hbio TableConfig{..} sessionRoot levels =
+openLevels reg hfs hbio conf@TableConfig{..} uc sessionRoot resolve levels =
     V.iforM levels $ \i -> openLevel (LevelNo (i+1))
   where
     mkPath = Paths.RunFsPaths (Paths.activeDir sessionRoot)
 
     openLevel :: LevelNo -> SnapLevel -> m (Level m h)
     openLevel ln SnapLevel{..} = do
-        incomingRuns <- openMergingRun snapIncomingRuns
+        (mmmay, incomingRuns) <- openMergingRun snapIncomingRuns
+        forM_ mmmay $ \(NumStepsDone c) -> supplyMergeCredits c incomingRuns -- TODO: this part is leaky!
```

Comment:
It is leaky if a merge completes, because the resulting run is not tracked in the `TempRegistry`. Closing input runs should also be tracked in the `TempRegistry`

Reply:
Moved this into a comment in the code.